selftests/unit/test_utils_asset.py

```diff
@@ -102,6 +106,32 @@ def test_unknown_scheme(self):
                               None, None, None, [self.cache_dir], None)
         self.assertRaises(asset.UnsupportedProtocolError, invalid.fetch)
 
+    def test_fetch_two_files_same_name_no_hash(self):
+        """
+        Checks that when two different assets which happen to have
+        the same *filename*, are properly stored in the cache
+        directory and that the right one will be given to the user,
+        no matter if a hash is used or not.
```

Comment:
well given the name the comment is misleading. It'll verify the right one will be given to the user even when no hash was supplied.

Reply:
OK, but the code says something different. The name is a bit misleading...

Reply:
I expected it to demonstrate that anonymous assets don't clash and I think it should do that. Would you please add third asset again without hash but with a different content? Because that is the motivation of this PR, isn't it?

Reply:
I mean
```diff
diff --git a/selftests/unit/test_utils_asset.py b/selftests/unit/test_utils_asset.py
index c2a3a848..79b0bd6e 100644
--- a/selftests/unit/test_utils_asset.py
+++ b/selftests/unit/test_utils_asset.py
@@ -126,12 +126,27 @@ class TestAsset(unittest.TestCase):
                          [self.cache_dir], None)
         a1.fetch()
         a2 = asset.Asset(second_asset_origin_url, None, None,
-                         [second_asset_origin_dir], [self.cache_dir],
+                         None, [self.cache_dir],
                          None)
         a2_path = a2.fetch()
         with open(a2_path, 'r') as a2_file:
             self.assertEqual(a2_file.read(), second_asset_content)
 
+        third_assetname = self.assetname
+        third_asset_origin_dir = tempfile.mkdtemp(dir=self.basedir)
+        third_asset_local_path = os.path.join(third_asset_origin_dir,
+                                              third_assetname)
+        third_asset_content = 'Another content!'
+        with open(third_asset_local_path, 'w') as f:
+            f.write(third_asset_content)
+        third_asset_origin_url = 'file://%s' % third_asset_local_path
+        a3 = asset.Asset(third_asset_origin_url, None, None,
+                         None, [self.cache_dir],
+                         None)
+        a3_path = a3.fetch()
+        with open(a3_path, 'r') as a3_file:
+            self.assertEqual(a3_file.read(), third_asset_content)
+
     def tearDown(self):
         shutil.rmtree(self.basedir)
```

Reply:
Yep, that can and should be done indeed.